engine/src/exec-files.cpp

```diff
@@ -254,9 +255,18 @@ void MCFilesEvalSpecialFolderPath(MCExecContext& ctxt, MCStringRef p_folder, MCS
 		return;
 	}
 
+    bool t_error;
     MCNewAutoNameRef t_path;
+    t_error = false;
     MCNameCreate(p_folder, &t_path);
-	if (MCS_getspecialfolder(*t_path, r_path))
+    // We have a special, mode-specific resource folder
+    if (MCStringIsEqualToCString(p_folder, "resources", kMCStringOptionCompareCaseless))
+        MCModeGetResourcesFolder(r_path);
+    if (!MCS_getspecialfolder(*t_path, r_path))
+        t_error = true;
+
```

Comment:
Doesn't this just overwrite the returned r_path from MCModeGetResourcesFolder with the one returned from MCS_getspecialfolder?